lib/ui/painting.dart

```diff
@@ -423,16 +419,19 @@ class Color {
       return false;
     }
     return other is Color &&
-        other.value == value &&
+        other.a == a &&
+        other.r == r &&
+        other.g == g &&
+        other.b == b &&
         other.colorSpace == colorSpace;
   }
 
   @override
-  int get hashCode => Object.hash(value, colorSpace);
+  int get hashCode => Object.hash(a, r, g, b, colorSpace);
 
-  // TODO(gaaclarke): Make toString() print out float values.
   @override
-  String toString() => 'Color(0x${value.toRadixString(16).padLeft(8, '0')})';
+  String toString() =>
```

Comment:
4 sig figs was chosen since hdmi 1.3 can support 12 bits per color component and 1/2^12 = 0.00024

Reply:
Scientific notation could have been another option if we wanted ( 2.4e-4 ).